Deployment/Jenkins/master/Master-Ec2-Instance.groovy

```diff
@@ -0,0 +1,281 @@
+pipeline {
+
+agent any
+
+options {
+    buildDiscarder(
+        logRotator(
+            numToKeepStr: '5',
+            daysToKeepStr: '30',
+            artifactDaysToKeepStr: '30',
+            artifactNumToKeepStr: '3'
+        )
+    )
+    disableConcurrentBuilds()
+    timeout(time: 60, unit: 'MINUTES')
+}
+
+environment {
+    AWS_DEFAULT_REGION = "${AwsRegion}"
+    AWS_CA_BUNDLE = '/etc/pki/tls/certs/ca-bundle.crt'
+    REQUESTS_CA_BUNDLE = '/etc/pki/tls/certs/ca-bundle.crt'
+}
+
+parameters {
+    string(name: 'AwsRegion', defaultValue: 'us-east-1', description: 'Amazon region to deploy resources into')
+    string(name: 'AwsCred', description: 'Jenkins-stored AWS credential with which to execute cloud-layer commands')
+    string(name: 'GitCred', description: 'Jenkins-stored Git credential with which to execute git commands')
+    string(name: 'GitProjUrl', description: 'SSH URL from which to download the Jenkins git project')
+    string(name: 'GitProjBranch', description: 'Project-branch to use from the Jenkins git project')
+    string(name: 'CfnStackRoot', description: 'Unique token to prepend to all stack-element names')
+    string(name: 'TemplateUrl', description: 'S3-hosted URL for the EC2 template file')
+    string(name: 'ElbName', description: 'ARN of the ELB to attach the EC2 instance')
+    string(name: 'AdminPubkeyURL', description: 'URL the file containing the admin users SSH public keys')
+    string(name: 'AmiDistro', defaultValue: 'CentOS', description: 'Linux distro of the AMI')
+    string(name: 'AmiId', description: 'ID of the AMI to launch')
+    string(name: 'AppVolumeDevice', defaultValue: '', description: 'Device to mount an extra EBS volume. Leave blank to launch without an extra application volume')
+    string(name: 'AppVolumeMountPath', defaultValue: '/var/lib/jenkins', description: 'Filesystem path to mount the extra app volume. Ignored if \"AppVolumeDevice\" is blank')
+    string(name: 'AppVolumeSize', defaultValue: '20', description: 'Size in GB of the EBS volume to create. Ignored if \"AppVolumeDevice\" is blank')
```

Comment:
Might want to check the space-utilization on the current master to see if 20GiB is going to be a sufficiently-large value (bearing in mind: the size of the current S3-hosted `sync` folder; the growth we've had to accommodate on the agents; the inbound influx of new plugins that will execute out of the master).

Reply:
I changed this default from 1 to 20 (just in case it was missed in the parameters). Planning on using 30 so I set the default to that in the pipeline file.